docs/proxy/datacenter_proxy/nodejs_examples.md

```diff
@@ -67,17 +68,20 @@ Keep a single IP selected from `SHADER` proxy group during the whole PuppeteerCr
     Apify.main(async () => {
         const requestList = new Apify.RequestList({ sources: [{ url: 'http://www.example.com' }] });
         await requestList.initialize();
+        const proxyConfiguration = await Apify.createProxyConfiguration({
+            // if you need to use more then one group
+            // simply add the additional ID's to the array below
+            groups: ['SHADER'],
+        });
 
         const crawler = new Apify.PuppeteerCrawler({
             requestList,
-            launchPuppeteerOptions: {
-                useApifyProxy: true,
-                // if you need to use more then one group
-                // simply add the additional ID's to the array below
-                apifyProxyGroups: ['SHADER'],
-                apifyProxySession: 'my_session',
+            proxyConfiguration,
+            useSessionPool: true,
+            sessionPoolOptions: {
+                sessionOptions: { maxPoolSize: 1 },
```

Comment:
We should mention in the example description that this does not keep the session for the whole crawl, but only until it fails (gets retired). The original solution would keep it forever (which sucked).

Reply:
The description was changed to `Keep a single IP selected from `SHADER` proxy group until it fails (gets retired)`.